src/_pytest/assertion/truncate.py

```diff
@@ -12,23 +12,40 @@
 
 
 DEFAULT_MAX_LINES = 8
-DEFAULT_MAX_CHARS = 8 * 80
+DEFAULT_MAX_CHARS = DEFAULT_MAX_LINES * 80
 USAGE_MSG = "use '-vv' to show"
 
 
-def truncate_if_required(
-    explanation: list[str], item: Item, max_length: int | None = None
-) -> list[str]:
+def truncate_if_required(explanation: list[str], item: Item) -> list[str]:
     """Truncate this assertion explanation if the given test item is eligible."""
     if _should_truncate_item(item):
-        return _truncate_explanation(explanation)
+        max_lines = int(
+            item.config.getini("truncation_limit_lines") or DEFAULT_MAX_LINES
+        )
+        max_chars = int(
+            item.config.getini("truncation_limit_chars") or DEFAULT_MAX_CHARS
+        )
+        return _truncate_explanation(
+            explanation,
+            max_lines=max_lines,
+            max_chars=max_chars,
+        )
     return explanation
 
 
 def _should_truncate_item(item: Item) -> bool:
     """Whether or not this test item is eligible for truncation."""
     verbose = item.config.get_verbosity(Config.VERBOSITY_ASSERTIONS)
-    return verbose < 2 and not util.running_on_ci()
+
```

Comment:
Yes, I also did not like it when I implemented this, sounds good to refactor this place :) 

Will think, what I can do here

However, I do think about return values here. `None` sounds great, but will it be clear that it means "No truncation" instead of using default values?

Reply:
> None sounds great, but will it be clear that it means "No truncation" instead of using default values?


You got a point, for a more visible API probably a new type would be best, but given this is internal and used in a single place, I guess using `None` is fine (as long as if documented of course).

Reply:
I've decided to return an additional `should_truncate` value. Seems simpler not to derive the truncation status somewhere else.